File: spring_crud/src/main/resources/app/dove/src/api/user/dto/UserLoginDto.ts
import {checks, Validate, String} from "@/util/Checks";

export interface UserLoginDto {
    password: string
    email: string
}

export function assertUserLoginDto (v: any): v is UserLoginDto {
    return new Validate({
        password: checks(
            String.min(8),
            String.max(500)
        ),
        email: String.max(255)
    }).check(v)
}

File: spring_crud/src/main/java/com/backend/dove/dto/UserLoginDto.java
package com.backend.dove.dto;

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import java.util.Collection;

public class UserLoginDto {

    @NotBlank
    @Max(500)
    @Min(8)
    String password;

    @NotBlank
    @Max(255)
    String email;

    public String getPassword() {
        return password;
    }

    public UserLoginDto setPassword(String password) {
        this.password = password;
        return this;
    }

    public String getEmail() {
        return email;
    }

    public UserLoginDto setEmail(String email) {
        this.email = email;
        return this;
    }
}
